tests/integration/test_source_git.py

```diff
@@ -641,3 +641,37 @@ def test_srpm_patch_non_conseq_indices(
         "0004-Wei-bier-Summer-is-coming.patch",
     }
     assert sg_path.joinpath(DISTRO_DIR, "saaz.patch").read_text() == ""
+
+
+def test_add_patch_with_patch_id(
+    mock_remote_functionality_sourcegit, api_instance_source_git
+):
+    """check that patches with patch_id set are added to spec correctly"""
+    spec_dir = api_instance_source_git.up.absolute_specfile_dir
+
+    # add a patch in there
+    spec: Specfile = api_instance_source_git.up.specfile
+
+    patch_name = "hello.patch"
+    # we need to create the patch file so that rebase-helper can find it and process it
+    patch_path = spec_dir.joinpath(patch_name)
+    patch_path.write_text("")
+    good_patch = PatchMetadata(
+        name=patch_name, path=patch_path, present_in_specfile=False, patch_id=10
```

Comment:
Also add a patch without the `patch_id` specified, to see that adding this will work.

Then add one patch without the `patch_id` specified after this (with `patch_id=10`) was added, to check that the `patch_id` of it is 11.

Reply:
okay